### PR TITLE
ba_contact_error_handling

### LINKED ISSUE #31: Error handling for returned phone numbers in action code
In `action_get_operator`, improve error handling for cases where zero or multiple entries are returned.

### PR DESCRIPTION
**Pull request by bjarkiarmanns**
Add error handling in cases where office contact info is not returned as it should.

Closes #31

### CODE REVIEW

**Comment by bjarkiarmanns** [comment]
Ready for review + merge, @bnika , @lumpidu

File: actions/actions.py
Comment (bnika) [inline actions/actions.py:289]: Am I missing something here, or would a "res" with either phone or email not be sufficient? If not phone AND mail, then "no-info"? - I see now where this comes from - but wouldn't still an option with either mail or phone be feasible?

**Comment by lumpidu** [comment]
My standard question: where is the test case ?

**Comment by bjarkiarmanns** [comment]
> My standard question: where is the test case ?

And the standard question is usually fair! But I'm not sure this calls for one - this only applies in case there is an error in retrieving the standard office contact info from our database, which shouldn't happen at all. Only if we should change our RDF setup at some future point in time, say. But I have tried doing that locally and this works.

**Comment by lumpidu** [comment]
Hmm, I not fully grasp it:

You are adding a new response message that is triggered either in case the API doesn't know a response for `get_office_contact_info()` or in case the telephone number or email is not set.

I know that a possibility for testing this is not straight forward with the current testing setup, but if such an error is not at all possible, then the whole MR is unnecessary. Maybe we should think about how this can be tested, so that we don't have untested error handling in the actions.

**Comment by bjarkiarmanns** [comment]
Right. Well maybe I misunderstood, I thought you were asking for this when the `ask_operator` intent was added, even though it was very unlikely we would get errors. I thought it was just to make sure our action code was not too custom-fitted to our current RDF. 

So if we want to drop this MR, rather than merge with no testing, we can do that. I really am not sure how we would test this exact scenario.

**Comment by lumpidu** [comment]
I would try to alter the entry of the knowledge base in a test case or replace the knowledge base altogether with one that triggers the error(s).

Dropping the MR is not the right approach. Whenever we would drop a MR because we need to add a testcase, then we'd do sth. seriously wrong :)

**Comment by lumpidu** [comment]
This should be testable now after #46 is merged. You just need to define an RDF file that triggers the edge cases and replace the original Fuseki DB with that RDF file before you test and restore the original Fuseki DB again afterwards so that the test order isn't a problem.

**Comment by bjarkiarmanns** [comment]
Ready for second review, @lumpidu , @bnika

**Review by bnika — DISMISSED** [review]

File: actions/actions.py
Comment (bjarkiarmanns) [reply actions/actions.py:289]: I've now added options for returning either a phone number or email and updated the tests accordingly.

**Review by bnika — APPROVED** [review]

**Comment by lumpidu** [comment]
lgtm

**Pull request merged**
